### DISCUSSION TITLE
Add an option to log the usage of the "Test user" feature

### DISCUSSION THREAD

**Discussion by andergmartins** [publishpress/PublishPress-Capabilities#678] [Feature Requests]
Adding a log allows admins to track who was testing user and when, so they can double check the log when any unusual behavior happened on the site to distinguish between the real user or any one else testing that user. Maybe only tracking when the admin click on test user, and when he logged back, and the direct logoff action as well, would be enough for that.

@stevejburge Maybe it could be a pro feature?

**Comment by stevejburge** [comment]
Thanks @andergmartins. Good idea. I wonder if we need a more general log of changes / usage in the future? Perhaps building on the backups we already store

**Comment by andergmartins** [comment]
That would be interesting in the future... but it would be a challenge to track general activities without impacting performance, because it would probably require hooking into a lot of additional actions or filters listening and recording changes. And how to track changes happened on 3rd party plugins without building specific listeners/hooks? That would probably require having a huge list of supported plugins. But might be possible for common changes in the WP core features at least.
